docker-examples/minimum-poetry/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3.11-slim
+
+ARG POETRY_VERSION=1.8
+
+# See https://python-poetry.org/docs/#ci-recommendations
+RUN pip install "poetry==${POETRY_VERSION}"
+
+WORKDIR /app
+
+# --- Reproduce the environment ---
+# You can comment the following two lines if you prefer to manually install
+#   the dependencies from inside the container.
+COPY pyproject.toml .
```

Comment:
Shouldn't we copy also copy `poetry.lock` for reproducibility?

Reply:
This is an important point.
Short answer. Yes, let's add it.

As much as I understand the function of the two (pyproject.toml, poetry.lock), in collaborative development, one can rely on even just the pyproject.toml to keep a general idea of the project's dependencies and configurations, in order to let the developers add what they need in terms of libraries and versions; moreover we can always get the latest versions at build time.
I didn't put the poetry.lock because this Dockerfile was for
> when you don't yet have in mind a clear idea of the environment requirements you need

In fact, on second thought, in this case, if we show a “minimal” example and want to take for granted the function of exactly reproducing an environment ,it becomes necessary. Thank you for pointing this out to me